File: backend/src/main/java/com/example/task_manager/task/dto/ExportTaskDTO.java
package com.example.task_manager.task.dto;

import com.example.task_manager.task.status.TaskStatus;
import lombok.Value;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

@Value
public class ExportTaskDTO {
    UUID id;
    String title;
    String description;
    TaskStatus status;
    Instant createdAt;
    Instant updatedAt;
    Instant dueDate;
    UUID createdById;
    String createdByUsername;
    List<UUID> assignedUserIds;
    List<String> assignedUsernames;
}

File: frontend/src/app/models/task/dto/export-task-dto.ts
export interface ExportTaskDTO {
  id: string;
  title: string;
  description: string;
  status: string;
  createdAt: string;
  updatedAt: string;
  dueDate: string;
  createdById: string;
  createdByUsername: string;
  assignedUserIds: string[];
  assignedUsernames: string[];
}
